include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -236,7 +236,7 @@ MatrixFree<dim, Number, VectorizedArrayType>::get_affine_constraints(
   AssertIndexRange(dof_handler_index, n_components());
 
   Assert(affine_constraints[dof_handler_index] != nullptr,
-         ExcMessage("AffineConstraints are only accessible if Number type "
```

Comment:
@kronbichler Should we switch to `AsserThrow`?

Reply:
I think `AssertThrow` would be good at this place indeed.